### ISSUE TITLE
OAuth 2.0 For Azure AD Authentication

### ISSUE THREAD

**Issue by brad55590** [marcusolsson/grafana-json-datasource#166]
Hi

We are looking at using your plugin to query one of our API's however our API uses Azure AD Authentication.

Could you please add an auth option for OAuth 2.0

Regards

**Comment by marcusolsson** [comment]
I will need to do some research before I can say, but it's definitely worth looking into 👍

**Comment by brad55590** [comment]
Found a possible solution/workaround:

By adding the following to the plugins.json file I am able to authenticate.

`"routes": [
    {
      "url": "",
      "tokenAuth": {
        "url": "https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token",
        "params": {
          "grant_type": "client_credentials",
          "client_id": "{clientId}",
          "client_secret": "{client_secret}",
          "scope": "{scope}"
        }
      }
    }
  ],`

Not sure if it is possible to add this as a configuration option?

**Comment by jaroli** [comment]
I am also trying to connect to a service with OAuth 2.0. The code @brad55590 posted works, but changing plugin.json invalidates the plugin signature.

**Comment by kopsor** [comment]
I agree with @jaroli. It would be better if there was a possibility to set the OAuth credentials in grafana data source GUI instead of doing a workaround and setting it in `plugin.json` file. It works but as mentioned it invalidates the plugin signature which then requires to: 
* remove plugin's `MANIFEST.txt` file 
* add this line `allow_loading_unsigned_plugins = marcusolsson-json-datasource` to `grafana.ini` config file 

in order to be able to use it.

**Comment by valdemarpavesi** [comment]
Is it possible to implement OAuth?

go to one URL get token by user/pass and assign token to base-URL-API.

example:
```
// get token
req_token, _ := http.NewRequest("POST", ".../identity/authenticate/oauth", bytes.NewBuffer(jsonStr_pw_user))
}
```

```
type
 AccessToken  string `json:"access_token"`
 RefreshToken string `json:"refresh_token"`
 TokenType    string `json:"token_type"`

 klog.V(2).Info(Tokenm.AccessToken)
```


add a token to a new request:
```
req, err := http.NewRequest("GET", ".../bla/bla-api", nil)
	if err != nil {
		klog.Error("Token, Error on response", err)
	}

req.Header.Add("authorization", "Bearer "+Tokenm.AccessToken)
req.Header.Add("Content-Type", "application/json;charset=UTF-8")
req.Header.Add("Accept", "application/json, text/plain, */*")
```

**Comment by marcusolsson** [comment]
I don't mind implementing this, but I'm struggling to find the time for it. If anyone wants to submit a PR for this, I'd be happy to review it. 

Heads-up though that I will be AFK for the next couple of weeks, and won't be able to review any PRs until I'm back 🙏

**Comment by valdemarpavesi** [comment]
thanks @marcusolsson 
I will study more the plugin.
